scripts/ai/strategies/AIDriveStrategyCourse.lua

```diff
@@ -703,4 +704,32 @@ function AIDriveStrategyCourse:updateInfoTexts()
             self:clearInfoText(infoText)
         end
     end
-end
\ No newline at end of file
+end
+
+------------------------------------------------------------------------------------------------------------------------
```

Comment:
Maybe we could move the detection phase into the Task logic? Could even create a new Task and but it in front of the working Tasks. This would be a bit cleaner I guess and also would reduce redundant code maybe.

Reply:
Dunno. There is no specific Task (or Job) for the Combine, everything is under CpAITaskFieldWork, but only the combine needs currently the field polygon. So that would have to be created. Also, Task to me is something that is useful and the driver has to do, not some internal activity, like detecting a field. Have no strong opinion it. I'd go ahead with what I have now, that is hopefully cleaner than it was before and then we can do another round.

Reply:
This seems to be a good plan than 👍